src/main/java/bio/terra/common/ValidationUtils.java

```diff
@@ -121,6 +163,11 @@ public static ArrayList<LinkedHashMap<String, String>> getRelationshipValidation
     if (toTerm != null) {
       errors.add(validateRelationshipTerm(toTerm, tables));
     }
+
+    if (fromTerm != null && toTerm != null) {
+      validateMatchingColumnDataTypes(fromTerm, toTerm, tables);
```

Comment:
This discards the errors, I think you want
```suggestion
      errors.add(validateMatchingColumnDataTypes(fromTerm, toTerm, tables));
```

Reply:
Thanks for catching this; I've also added test coverage.